File: flutter_application_1/lib/models/work_shift.dart
class WorkShift {
  final int id;
  final String shiftCode;
  final String shiftName;
  final String startTime;
  final String endTime;
  final int maxPatients;
  final bool status;

  WorkShift({
    required this.id,
    required this.shiftCode,
    required this.shiftName,
    required this.startTime,
    required this.endTime,
    required this.maxPatients,
    required this.status,
  });

  factory WorkShift.fromJson(Map<String, dynamic> json) {
    return WorkShift(
      id: json['id'],
      shiftCode: json['shift_code'],
      shiftName: json['shift_name'],
      startTime: json['start_time'],
      endTime: json['end_time'],
      maxPatients: json['max_patients'],
      status: json['status'],
    );
  }
}

File: Backend/models/work_shift.py
from sqlalchemy import Column, BigInteger, String, Time, Integer, Boolean, DateTime
from sqlalchemy.sql import func

from db import Base


class WorkShift(Base):
    __tablename__ = "work_shifts"

    id = Column(BigInteger, primary_key=True, index=True)
    shift_code = Column(String(20), unique=True, nullable=False)
    shift_name = Column(String(100), nullable=False)

    start_time = Column(Time, nullable=False)
    end_time = Column(Time, nullable=False)

    max_patients = Column(Integer, default=20)

    status = Column(Boolean, default=True)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )

    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now()
    )